### DISCUSSION TITLE
Is there any alternative to npx in bun?

### DISCUSSION THREAD

**Discussion by KeshariPiyush24** [oven-sh/bun#816] [Q&A] [answered]
I would like to run `npx kill-port 3000` command in bun. How to do that?

**Comment by Jarred-Sumner** [comment, marked as answer]
~~Not yet, but really should add that~~

~~The closest is `bun add -g` so for your case, you could do `bun add -g kill-port` and then run `kill-port 3000`~~

You can use `bunx`

```js
bunx fkill-cli 3000
```

**Reply by cyfung1031** [reply]
@Jarred-Sumner global package installation does not work in Linux and macOS as the installation of bun would not add `BUN_INSTALL` and `PATH` automatically #5990 

<img width="563" alt="Screen Shot 2023-09-24 at 11 10 58" src="https://github.com/oven-sh/bun/assets/44498510/9bf0b1fb-c3e9-4361-9445-fd5ffb7c64e6">

**Comment by ngraf** [comment]
By now there is `bunx` , see https://bun.sh/docs/cli/bunx
> Use bunx to auto-install and run packages from npm. It's Bun's equivalent of npx or yarn dlx.

Example: `bunx cowsay "Hello world!"`

**Comment by palashkaria** [comment]
I came here looking for exactly this, hoping to make `kill-port` faster :) ah well

**Reply by palashkaria** [reply]
If anyone comes here looking to make kill-port faster: this worked for me (just revert to 1.6.1)

> For now, i rollback to prev version:
> npm install --save kill-port@1.6.1



https://github.com/tiaanduplessis/kill-port/issues/61#issuecomment-1133860058

**Comment by micalevisk** [comment]
You can see on their source code that it has a shebang `#!/usr/bin/env node`

So I believe that will use nodejs, in the end.

**Reply by marouane-ach** [reply]
You can use the [--bun flag](https://bun.sh/docs/cli/bunx#shebangs)
